### ISSUE TITLE
this.googleApi.onLoad(...).mergeMap is not a function

### ISSUE THREAD

**Issue by Xtremrules** [rubenCodeforges/ng-gapi#63]
This is the log of the error I receive.
```
HomeComponent.html:2 ERROR TypeError: this.googleApi.onLoad(...).mergeMap is not a function
    at GoogleAuthService.push../node_modules/ng-gapi/lib/GoogleAuthService.js.GoogleAuthService.getAuth (GoogleAuthService.js:18)
    at GoogleSignInService.push../src/app/shared/google-sign-in.service.ts.GoogleSignInService.signIn (google-sign-in.service.ts:36)
    at HomeComponent.push../src/app/home/home.component.ts.HomeComponent.signIn (home.component.ts:26)
    at Object.eval [as handleEvent] (HomeComponent.html:2)
    at handleEvent (core.js:10258)
    at callWithDebugContext (core.js:11351)
    at Object.debugHandleEvent [as handleEvent] (core.js:11054)
    at dispatchEvent (core.js:7717)
    at core.js:8161
    at HTMLAnchorElement.<anonymous> (platform-browser.js:995)
```
The error seems to be coming from Here
```
GoogleAuthService.prototype.getAuth = function () {
        var _this = this;
        if (!this.GoogleAuth) {
            return this.googleApi.onLoad().mergeMap(function () { return _this.loadGapiAuth(); });
        }
        return Observable.of(this.GoogleAuth);
    };
```

**Comment by Xtremrules** [comment]
This is the content of `package.json`
```
{
  "name": "pgsadmin-ui",
  "version": "0.0.0",
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "build": "ng build",
    "test": "ng test",
    "lint": "ng lint",
    "e2e": "ng e2e"
  },
  "private": true,
  "dependencies": {
    "@angular/animations": "^6.1.6",
    "@angular/common": "^6.1.0",
    "@angular/compiler": "^6.1.0",
    "@angular/core": "^6.1.0",
    "@angular/forms": "^6.1.0",
    "@angular/http": "^6.1.0",
    "@angular/platform-browser": "^6.1.0",
    "@angular/platform-browser-dynamic": "^6.1.0",
    "@angular/router": "^6.1.0",
    "core-js": "^2.5.4",
    "crypto-js": "^3.1.9-1",
    "lodash": "^4.17.10",
    "ng-gapi": "0.0.51",
    "ngx-toastr": "^9.0.2",
    "rxjs": "^6.0.0",
    "zone.js": "~0.8.26"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "~0.7.0",
    "@angular/cli": "~6.1.5",
    "@angular/compiler-cli": "^6.1.0",
    "@angular/language-service": "^6.1.0",
    "@types/jasmine": "~2.8.6",
    "@types/jasminewd2": "~2.0.3",
    "@types/node": "~8.9.4",
    "codelyzer": "~4.2.1",
    "jasmine-core": "~2.99.1",
    "jasmine-spec-reporter": "~4.2.1",
    "karma": "~1.7.1",
    "karma-chrome-launcher": "~2.2.0",
    "karma-coverage-istanbul-reporter": "~2.0.0",
    "karma-jasmine": "~1.1.1",
    "karma-jasmine-html-reporter": "^0.2.2",
    "protractor": "~5.4.0",
    "ts-node": "~5.0.1",
    "tslint": "~5.9.1",
    "typescript": "~2.7.2"
  }
}
```

**Comment by Xtremrules** [comment]
I updated all npm packages and installed
```
"rxjs": "^6.3.2",
"rxjs-compat": "^6.3.2",
``
I the issue still persist.

**Comment by rubenCodeforges** [comment]
try 
observable.pipe(mergeMap())

**Comment by rubenCodeforges** [comment]
@Xtremrules i will update the lib , looks like compat is not more working

**Comment by projectzerotohero** [comment]
Almost same issue with ours that we just noticed today.

Seems like the part that has ` this.googleService.onLoad()` causes the issue.


Can you recommend any workarounds?

**Comment by theNotoriousHxc** [comment]
@rubenCodeforges I have the same issue with @projectzerotohero

**Comment by stevecances** [comment]
In the meantime, is their any workaround ?
@rubenCodeforges  @projectzerotohero

**Comment by stevecances** [comment]
Yes it works with `pipe(mergeMap())`.
My workaround is to provide my [own implementation of `GoogleAuthService` with the fix](https://gist.github.com/stevecances/2a4aad4d1c0d255622cd5d5f49ad3cc4).

**Comment by rubenCodeforges** [comment]
So guys this is a issue with RXJS 6.0 , they have deprecated the old way of chaining operators. Please see the change notes of RxJs 6.0.

You need to use `pipe(mergeMap())`

**Comment by shanepisko** [comment]
@rubenCodeforges  where do we need to change the pip(mergeMap()) ? i tried editing in the ng-gapi/lib/GoogleAuthService.js but I now get an error "mergemap is not defined" after switching that line to this: in the get auth method: 
`  GoogleAuthService.prototype.getAuth = function () {
        var _this = this;
        if (!this.GoogleAuth) {
            return this.googleApi.onLoad().pipe(mergeMap(function () { return _this.loadGapiAuth(); }));
        }
        return Observable.of(this.GoogleAuth);
    };`

**Comment by shanepisko** [comment]
I followed @stevecances own implementation and swapped that for what was in the GoogleAuthService.prototype.getAuth method, and that worked perfectly.
`
GoogleAuthService.prototype.getAuth = function () {
        var _this = this;
        if (!this.GoogleAuth) {
            return this.googleApi.onLoad().pipe(mergeMap(() => this.loadGapiAuth()));
        }
        return Observable.of(this.GoogleAuth);
    };
`

**Comment by rubenCodeforges** [comment]
fixed in v0.0.6 see release notes https://github.com/rubenCodeforges/ng-gapi/releases/tag/v0.0.6
